MaxText/train.py

```diff
@@ -196,33 +197,55 @@ def train_step(model, config, state, data, dropout_rng):
   rng1, aqt_rng = jax.random.split(dropout_rng)
 
   # decimate proportion of data when per_device_batch_size<1
-  for k, v in data.items():
-    data[k] = v[:config.global_batch_size_to_train_on,:]
+  if is_train:
+    for k, v in data.items():
+      data[k] = v[:config.global_batch_size_to_train_on,:]
 
-  def loss_fn(params):
+  def loss_fn(params, is_train=True):
     logits, intermediate_outputs = model.apply({'params': params},
                          data['inputs'],
                          data['inputs_position'],
                          decoder_segment_ids=data['inputs_segmentation'],
-                         enable_dropout=config.enable_dropout,
+                         enable_dropout=config.enable_dropout if is_train else False,
```

Comment:
Why? Can't you set Dropout= 0?

Reply:
Hi, @rwitten 

`Dropout = 0`(`dropout_rate=0`) is equivalent to `enable_dropout = False`, while `model` allows us to pass `enable_dropout` only and `dropout_rate` is fixed in config. Considering we enable dropout with a valid `dropout_rate` in training while disabling it evaluation.

Looking into the code in depth, `deterministic` is simply designed to disable dropout in evaluation:
```
# flax https://flax.readthedocs.io/en/latest/_modules/flax/linen/stochastic.html#Dropout
x = nn.Dropout(0.5, deterministic=not train)(x)

# ours
deterministic=not enable_dropout,
...
nn.Dropout(
        rate=cfg.dropout_rate, broadcast_dims=(-2,)
    )(x, deterministic=deterministic)
```

Reply:
yes this makes sense.